internal/agent/agent.go

```diff
@@ -86,7 +94,7 @@ func (a *Agent) Run(ctx context.Context) error {
 
 	// initialize the TPM
 	var tpmChannel *tpm.TPM
-	if len(a.config.TPMPath) > 0 {
+	if a.config.TPMPath == "" {
```

Comment:
Why did we change this condition? shouldn't it stay as before?

Reply:
This is another topic, but I suspect we should try to default to /dev/tpm0 or the right one... if it fails for no tpm may be the user can disable it.

Reply:
just a style change

Reply:
> This is another topic, but I suspect we should try to default to /dev/tpm0 or the right one... if it fails for no tpm may be the user can disable it.

Yeah I feel this is not related to this PR but a good topic.

Reply:
> Why did we change this condition? shouldn't it stay as before?
good catch I think I was looking at adjust the logic but was not completed.

Reply:
done

Reply:
tracked here https://github.com/flightctl/flightctl/issues/170